src/scripts/towns/BattleCafe.ts

```diff
@@ -50,21 +50,26 @@ class BattleCafeController {
 
         BattleCafeController.clockwise(clockwise);
         BattleCafeController.isSpinning(true);
-        BattleCafeController.spinsLeft(BattleCafeController.spinsLeft() - 1);
         const spinTime = +$('#battleCafeDuration').val();
         const sweet = BattleCafeController.selectedSweet();
-        BattleCafeController.getPrice(sweet).forEach(b => GameHelper.incrementObservable(App.game.farming.berryList[b.berry], b.amount * -1));
+
 
         setTimeout(() => {
             BattleCafeController.isSpinning(false);
             BattleCafeController.unlockAlcremie(clockwise, spinTime, sweet);
+            BattleCafeController.spinsLeft(BattleCafeController.spinsLeft() - 1);
+            BattleCafeController.getPrice(sweet).forEach(b => GameHelper.incrementObservable(App.game.farming.berryList[b.berry], b.amount * -1));
         },
         spinTime * 1000);
     }
 
     private static unlockAlcremie(clockwise: boolean, spinTime: number, sweet: GameConstants.AlcremieSweet) {
         let spin: GameConstants.AlcremieSpins;
         const curHour = (new Date()).getHours();
+        if (spinTime == 3600) {
+            (new PokemonItem('Milcery (Cheesy)', 0)).gain(1);
+            return;
+        }
         if (curHour == 18 && !clockwise && spinTime > 10) {
             spin = GameConstants.AlcremieSpins.at7Above10;
```

Comment:
Shouldn't this be `at6Above10?` or `curHour == 19`?

Reply:
You are right. Nice catch!